### PR TITLE
Add AVL tree rotation example image;

### PR DESCRIPTION
**Pull request by Antondomashnev**
I've tried to follow the style of existing images in the AVL tree readme, however it's not 100% the same 😄

### CODE REVIEW

**Comment by kelvinlauKL** [comment]
I think it'll be good to explain the steps in the rotation. For example, how you might find the pivot, how you want to disconnect and reconnect the nodes. 

Otherwise, the graph is a solid addition 👍

**Comment by Antondomashnev** [comment]
@kelvinlauKL Thanks for the response, I'm completely agree, I'll update the PR with the steps description 👍

**Comment by Antondomashnev** [comment]
@kelvinlauKL I've added some explanation, could you take a look if this made the image clear or not 😄

File: AVL Tree/README.markdown [hunk 1]
```diff
@@ -43,13 +43,32 @@ The difference between the heights of the left and right subtrees is called the
...
-
 Each tree node keeps track of its current balance factor in a variable. After inserting a new node, we need to update the balance factor of its parent node. If that balance factor becomes greater than 1, we "rotate" part of that tree to restore the balance.
 
-Example of balancing the unbalanced tree using rotation:
+Example of balancing the unbalanced tree using *Right* (clockwise direction) rotation :
 ![Rotation](Images/Rotation.jpg)
 
-Insertion never needs more than 2 rotations. Removal might require up to *log(n)* rotations.
+Let's dig into rotation algorithm in detail using the terminology:
+* *Root* - the parent not of the subtrees that will be rotated;
```
Comment (kelvinlauKL) [inline AVL Tree/README.markdown]: typo the "the parent not" should be "the parent node.

File: AVL Tree/README.markdown [hunk 2]
```diff
@@ -43,20 +43,40 @@ The difference between the heights of the left and right subtrees is called the
...
 ## Rotations
-
 Each tree node keeps track of its current balance factor in a variable. After inserting a new node, we need to update the balance factor of its parent node. If that balance factor becomes greater than 1, we "rotate" part of that tree to restore the balance.
 
-TODO: describe with pictures how these rotations work
+Example of balancing the unbalanced tree using *Right* (clockwise direction) rotation :
+![Rotation](Images/Rotation.jpg)
+
+Let's dig into rotation algorithm in detail using the terminology:
+* *Root* - the parent not of the subtrees that will be rotated;
```
Comment (kelvinlauKL) [inline AVL Tree/README.markdown]: typo here - "parent not" should be "parent node"

**Comment by Antondomashnev** [comment]
@kelvinlauKL thanks for the valuable feedback, I'll work on improvements 👍

**Comment by Antondomashnev** [comment]
@kelvinlauKL please ignore the commits for now, we can squash them later 😄

**Comment by kelvinlauKL** [comment]
Okie dokie, let me know when it's ready.

**Comment by Antondomashnev** [comment]
> Okie dokie, let me know when it's ready.

Ah sorry, I meant ignore the commit message and that the message is repeating 😄 but you can take a look on the implementation already 😉

**Comment by kelvinlauKL** [comment]
Looks great 👍 👍 👍 Merging it in. Thanks @Antondomashnev, and Merry Christmas!!

**Comment by Antondomashnev** [comment]
@kelvinlauKL merry Christmas ⛄️

**Pull request merged**